docs/topics/upgrade.md

```diff
@@ -0,0 +1,77 @@
+# Upgrading Kubernetes Clusters
+
+## Prerequisites
+
+All the commands in this guide require both the Azure CLI and `aks-engine`. Follow the [quickstart guide](../tutorials/quickstart.md) before continuing.
+
+This guide assumes you already have deployed a cluster using `aks-engine`. For more details on how to do that see [deploy](../tutorials/deploy.md).
+
+## Upgrade
+
+This document provides guidance on how to upgrade the Kubernetes version for an existing `aks-engine` cluster and recommendations for adopting `aks-engine upgrade` as a production quality tool.
+
+<a name="pre-requirements"></a>
+
+### Know before you go
+
+In order to ensure that your `aks-engine upgrade` operation runs smoothly, there are a few things you should be aware of before getting started.
+
+1) You will need access to the original `_output/` directory files that were generated during cluster deployment in order to run `upgrade`.  `aks-engine` will use the `--deployment-dir` argument provided in the upgrade command to introspect the Azure Resource Manager (ARM) template and parameters to determine the cluster's current Kubernetes version and its current state.
+
+2) `aks-engine upgrade` expects a cluster configuration that conforms to the current state of the cluster. In other words, the Azure resources deployed by `aks-engine` should be in the same state that when they were originally created by `aks-engine`. If you perform manual operations on your Azure IaaS resources (other than `aks-engine scale` and `aks-engine upgrade`) you risk not being able to upgrade your cluster successfully as `aks-engine` won't be able to reconcile the cluster's current state. This includes naming of resources; `aks-engine` relies on some resources (such as VMs) to be named in accordance with the original `aks-engine` deployment.
```

Comment:
How about this for additional literal accuracy:

"`aks-engine upgrade` expects a cluster configuration that will produce ARM templates that can be deployed as an incremental deployment on top of the existing resource group" (<-- validate that this is true, ref: https://docs.microsoft.com/en-us/azure/azure-resource-manager/deployment-modes)

Basically, let's try to succinctly state was is actually happening. aks-engine is a convenience that is implementing an ARM template deployment against a resource group that has been created in the past as a result of a prior ARM template deployment.

Reply:
"...you risk not being able to upgrade your cluster successfully, as the `aks-engine`-generated ARM template won't be reconcilable against the state of the Azure resources that reside in the resource group."

Also maybe add something like this to the end:

"In summary, the set of Azure resources in the resource group are mutually reconcilable only if they have been exclusively created and managed as the result of a series of successive ARM template deployments originating from `aks-engine`."

Reply:
I don't understand your first comment "Basically, let's try to succinctly state was is actually happening"